File: src/main/java/ru/nsu/fit/team_project/model/commands/SetFieldValueCommand.java
package ru.nsu.fit.team_project.model.commands;

import ru.nsu.fit.team_project.model.Model;
import ru.nsu.fit.team_project.model.fields.Field;

import java.util.UUID;

public class SetFieldValueCommand extends Command {
    private final UUID fieldID;
    private final Object value;

    public SetFieldValueCommand(UUID fieldID, Object value) {
        super("SetFieldValueCommand");
        this.fieldID = fieldID;
        this.value = value;
    }

    @Override
    public void execute(Model model) {
        Field field = model.getObjects()
                           .stream()
                           .filter(o -> o.getFields()
                                         .stream()
                                         .anyMatch(f -> f.getID().equals(fieldID))
                           )
                           .findAny()
                           .orElseThrow()
                           .getField(fieldID);
        field.setValue(value);
    }
}

File: src/jvmMain/kotlin/data/model/commands/SetFieldValueCommand.kt
package data.model.commands

import data.model.Model
import java.util.*

class SetFieldValueCommand(
    private val fieldId: UUID,
    private val value: Any
) : Command(COMMAND_NAME) {

    private companion object {

        private const val COMMAND_NAME = "SetFieldValue"
    }

    override fun execute(model: Model) {
        model.objects
            .stream()
            .filter { o -> o.fields
                    .stream()
                    .anyMatch { f -> f.id == fieldId }
            }
            .findAny()
            .orElseThrow()
            .getFieldById(fieldId)
            .value = value

    }
}